dapp/src/components/dialogs/DeleteNameDialog.tsx

```diff
@@ -0,0 +1,143 @@
+    Dialog,
+    DialogBody,
+    DialogContent,
+    Header,
+    InfoBox,
+    InfoBoxStyle,
+    InfoBoxType,
+    LoadingIndicator,
+} from '@iota/apps-ui-kit';
+import { useCurrentAccount, useIotaClient, useSignAndExecuteTransaction } from '@iota/dapp-kit';
+import { isSubName } from '@iota/iota-names-sdk';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+
+import { useRegistrationNfts } from '@/hooks';
+import { queryKey } from '@/hooks/queryKey';
+import { NameUpdate, useUpdateNameTransaction } from '@/hooks/useUpdateNameTransaction';
+import { RegistrationNft } from '@/lib/interfaces/registration.interfaces';
+import { getNameObject } from '@/lib/utils/names';
+
+type DeleteNameDialogProps = {
+    nft: RegistrationNft;
+    open: boolean;
+    setOpen: (bool: boolean) => void;
+};
+
+export function DeleteNameDialog({ nft, open, setOpen }: DeleteNameDialogProps) {
+    const iotaClient = useIotaClient();
+    const queryClient = useQueryClient();
+
+    const account = useCurrentAccount();
+    const isConnected = !!account?.address;
+    if (!isConnected) return null;
+
+    const { data: domainsOwned } = useRegistrationNfts('domain');
+    const { data: subdomainsOwned } = useRegistrationNfts('subdomain');
+
+    const isNameSubName = nft ? isSubName(nft.name) : null;
+    const isExpired = nft.isExpired || false;
+
+    // Create updates
+    const updates: NameUpdate[] = [];
+
+    if (nft && isExpired) {
+        const nftId = getNameObject(domainsOwned ?? [], subdomainsOwned ?? [], nft.name);
+        if (nftId) {
+            updates.push({
+                type: 'delete-name',
+                nft: nftId,
+                isSubname: isNameSubName || false,
+            });
+        }
+    }
+    const {
+        data: updateNameTransaction,
+        error: updateNameError,
+        isLoading: isLoadingUpdateNameTransaction,
+    } = useUpdateNameTransaction({
+        address: account?.address || '',
+        updates,
+        isExpired,
+    });
+
+    const { mutateAsync: signAndExecuteTransaction, isPending: isSendingTransaction } =
+        useSignAndExecuteTransaction();
+
+    const { mutate: save, isPending: isSaving } = useMutation({
+        async mutationFn() {
+            if (!updateNameTransaction) return;
+            const transaction = await signAndExecuteTransaction({
+                transaction: updateNameTransaction,
+            });
+            await iotaClient.waitForTransaction({
+                digest: transaction.digest,
+            });
+            for (const update of updates) {
+                switch (update.type) {
+                    case 'delete-name':
+                        queryClient.invalidateQueries({
+                            queryKey: queryKey.ownedObjects(account?.address || ''),
+                        });
+                        break;
+                }
+            }
+        },
+        onSuccess: () => {
+            closeDialog();
+        },
```

Comment:
No need for the switch in this hook, its not like you can do anything else besides deleting, you can also just move the invalidation to the onSuccess
```suggestion
        },
        onSuccess: () => {
        	queryClient.invalidateQueries({
                queryKey: queryKey.ownedObjects(account?.address || ''),
            });
            closeDialog();
        },
```